src/mappings/util.ts

```diff
@@ -138,28 +149,92 @@ export const getSigner = (extrinsic: SubstrateExtrinsic): string => {
   );
 };
 
-export const harvesterLikeParamsToObj = (
-  params: HarvesterLikeCallArgs,
-  formatKey = true
-): Record<string, any> => {
-  const obj: Record<string, any> = {};
-  params.forEach(p => {
-    obj[formatKey ? snakeToCamelCase(p.name) : p.name] =
-      p.name === 'asset_type' ? Object.keys(p.value)[0] : p.value;
-  });
-  return obj;
+/**
+ * Parses a raw Asset Document
+ */
+export const getDocValue = (
+  doc: Codec
+): Pick<AssetDocument, 'name' | 'link' | 'contentHash' | 'type' | 'filedAt'> => {
+  const {
+    uri: link,
+    content_hash: documentHash,
+    name,
+    doc_type: type,
+    filing_date: filedAt,
+  } = JSON.parse(doc.toString());
+
+  const hashType = Object.keys(documentHash)[0];
+
+  return {
+    name,
+    link,
+    contentHash: documentHash[hashType],
+    type,
+    filedAt,
+  };
 };
 
-export const formatAssetIdentifiers = (
-  identifiers: Record<string, string>[]
-): SecurityIdentifier[] =>
-  identifiers.map(i => {
+export const getSecurityIdentifiers = (item: Codec): SecurityIdentifier[] => {
+  const identifiers = JSON.parse(item.toString());
+  return identifiers.map(i => {
     const type = Object.keys(i)[0];
     return {
       type,
       value: i[type],
     };
   });
+};
+
+/**
+ * Parses a Vec<AssetCompliance>
+ */
+export const getComplianceRulesValue = (
+  requirements: Codec
+): Pick<Compliance, 'complianceId' | 'data'>[] => {
+  return JSON.parse(JSON.stringify(requirements));
+};
+
+/**
+ * Parses AssetCompliance
+ */
+export const getComplianceValue = (
+  compliance: Codec
+): Pick<Compliance, 'complianceId' | 'data'> => {
+  const { id, ...data } = JSON.parse(compliance.toString());
+  return {
+    complianceId: Number(id),
+    data,
+  };
+};
+
+/**
+ * Parses AssetTransferManager
+ */
+export const getTransferManagerValue = (
+  manager: Codec
+): Pick<TransferManager, 'type' | 'value'> => {
+  const { countTransferManager, percentageTransferManager } = JSON.parse(JSON.stringify(manager));
```

Comment:
5.0.0 renamed these to be "Conditions" instead of "Manager". Try to test these against 5.0.0

We might need to support both names for 4.0.0 and 5.0.0 chains.

Long term I think just the new name will be OK since the storage should get migrated on chain.